plugins/woocommerce-blocks/assets/js/base/components/state-input/state-input.tsx

```diff
@@ -34,18 +41,37 @@ const StateInput = ( {
 	autoComplete = 'off',
 	value = '',
 	required = false,
+	errorId,
 }: StateInputWithStatesProps ): JSX.Element => {
 	const countryStates = states[ country ];
-	const options = useMemo(
-		() =>
-			countryStates
-				? Object.keys( countryStates ).map( ( key ) => ( {
-						value: key,
-						label: decodeEntities( countryStates[ key ] ),
-				  } ) )
-				: [],
-		[ countryStates ]
-	);
+	const options = useMemo< SelectOption[] >( () => {
+		if ( countryStates && Object.keys( countryStates ).length > 0 ) {
```

Comment:
In some cases countryStates is an empty array causing us to add a select with an empty option. This fixes that case.